contracts/liquidity_pool.cairo

```diff
@@ -635,6 +661,124 @@ func _get_value_of_pool_position{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*,
 }
 
 
+@view
+func get_user_poolinfo{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+} (user: Address) -> (
+    upoolinfo_len: felt,
+    upoolinfo: UserPoolInfo*
+) {
+    alloc_locals;
+    let (lptoken_addrs_len: felt, lptoken_addrs: Address*) = get_all_lptoken_addresses();
+    let (res: UserPoolInfo*) = alloc();
+    let (upoolinfo_len: felt) = map_and_filter_address_to_userpoolinfo(user, lptoken_addrs, lptoken_addrs_len, res);
+    return (upoolinfo_len, res);
+}
+
+func map_and_filter_address_to_userpoolinfo{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+} (user_addr: Address, lptoken_addrs: Address*, lptoken_addrs_len: felt, upoolinfo: UserPoolInfo*)
+    -> (upoolinfo_len: felt){
+    if (lptoken_addrs_len == 0){
+        return (0,);
+    }
+    let val = get_userpoolinfo(user_addr, lptoken_addrs[0]);
```

Comment:
we have two function
- get_userpoolinfo
- get_user_poolinfo
with only the underscore as to differentiate... could you rename one of them?